src/System.Memory/tests/TestHelpers.cs

```diff
@@ -20,7 +20,7 @@ public static class TestHelpers
             Assert.True(span.SequenceEqual(expected));
         }
 
-        public static void ValidateReferenceType<T>(this Span<T> span, params T[] expected)
+        public static void ValidateReferenceType<T>(this Span<T> span, params T[] expected) where T : class
```

Comment:
Regarding the new generic constraints on a few methods: xUnit analyser complained that I'm supposed to use `Assert.Equal` instead of `Assert.Same` for value type `T`s. Since this was for reference types anyway (`*Reference`), I limited the `T`s to be reference types only.

Reply:
Thanks for the explanation here 👍